main/lsp/LSPTypechecker.cc

```diff
@@ -191,11 +271,10 @@ bool LSPTypechecker::runSlowPath(LSPFileUpdates updates, WorkerPool &workers, bo
     finalGS->errorQueue = make_shared<core::ErrorQueue>(finalGS->errorQueue->logger, finalGS->errorQueue->tracer);
     finalGS->errorQueue->ignoreFlushes = true;
     auto &epochManager = *finalGS->epochManager;
-    // TODO: Replace with an actual preemption task manager when we ship preemptible slow path.
-    optional<shared_ptr<core::lsp::PreemptionTaskManager>> preemptManager;
+    const u4 epoch = updates.epoch;
     // Note: Commits can only be canceled if this edit is cancelable, LSP is running across multiple threads, and the
     // cancelation feature is enabled.
-    const bool committed = epochManager.tryCommitEpoch(updates.epoch, cancelable, preemptManager, [&]() -> void {
+    const bool committed = epochManager.tryCommitEpoch(*finalGS, epoch, cancelable, preemptManager, [&]() -> void {
```

Comment:
The insidious bug that I had is that I was passing `gs` to `tryCommitEpoch` rather than `finalGS`, so it wasn't replacing the `errorQueue` during preemption at the end of a slow path. Thus, a fast path would come through, drain the slow path's errors, and then the slow path would complete thinking all errors have been fixed.